docs/design.md

```diff
@@ -0,0 +1,59 @@
+
+As a result, the non-workload components running inside enclave are not mandatory to be provided by image creator. Meanwhile, this approach can facilitate the development of standardization of implementing Enclave-CC, which reuses the methodology of image protections already applied to the container ecosystem, such as image [encryption](https://github.com/containers/ocicrypt-rs) and [signing](https://github.com/containers/image/blob/main/docs/containers-signature.5.md), instead of using LibOSes specific approaches.
+
+![no_wrapping.png](docs/images/no_wrapping.png)
+
+### Align with Confidential Containers principles
+
+Confidential Containers design principle tends to utilize and reuse ecosystem to implement Confidential Containers. More specifically, Confidential Containers requires:
+
+- Container image is always protected, e.g, signed and/or encrypted.
+- Download and unpack container image to the secure and attested enclave for the actual workloads to run with the protections of confidentiality and integrity.
+- Prove that sensitive workloads are running on a genuine and trusted hardware TEE.
+- Follow up trust and threat model defined by Confidential Containers.
+
+Enclave-CC approach would strictly align with above design principles to implement a general process-based Confidential Containers co-existing with Pod-based Confidential Containers. It provides a consistent Kubernetes user experience across hardware TEEs, but also compared with the existing, non confidential Kubernetes usage patterns. The container workload is unmodified (but encrypted/signed) and identical across hardware TEEs.
+
+![k8s_ux.png](docs/images/k8s_ux.png)
+
+where:
+- [Inclavare Containers Project](https://github.com/inclavare-containers/inclavare-containers) is a container runtime to launch the process-based Confidential Container for Intel SGX.
+
+- LibOS generally refers to the SGX LibOS implementations, such as [Gramine](https://github.com/gramineproject/gramine) and [Occlum](https://github.com/occlum/occlum).
+
+- `rune` is the container runtime for spawning and running enclaves in containers.
+
+- `runelet` is the containerized process used to interact with LibOS through common PAL APIs.
+
+## Architecture
+
+![arch.png](docs/images/arch.png)
+
+Enclave-CC introduces a helper enclave called `stub enclave` which is a fixed function enclave. It is launched by `shim-rune` during pod creation in a private method. More specifically, `stub enclave` is deployed in the form of OCI bundle rather than a container image, and thus `shim-rune` can call `rune` to consume it. `stub enclave` is kept the same life cycle as pod, waiting to load the image into protected enclave memory.  Be aware of that `stub enclave` must be attested by verifier before launching any `app enclaves`. The attestation to `stub enclave` will be triggered by the decryption of `app image`.
```

Comment:
```suggestion
Enclave-CC introduces a helper enclave called `stub enclave` which is a fixed function enclave. It
is launched by `shim-rune` during pod creation in a private method. More specifically, `stub
enclave` is deployed in the form of OCI bundle rather than a container image, and thus `shim-rune`
can call `rune` to start it. `stub enclave` has the same life cycle as the pod. Its main job is to
receive and then load the actual workload's container image into an `app enclave`.  
To do this securely both the stub and app enclaves establish trust through attestation.
```